unstructured/ingest/connector/mongodb.py

```diff
@@ -78,39 +88,203 @@ def to_dict(
+        return {
+            "host": self.connector_config.host,
+            "collection": self.connector_config.collection,
+            "document_id": self.document_meta.document_id,
+        }
+
+
+@dataclass
+class MongoDBIngestDocBatch(BaseIngestDocBatch):
+    connector_config: SimpleMongoDBConfig
+    ingest_docs: t.List[MongoDBIngestDoc] = field(default_factory=list)
+    list_of_ids: t.List[str] = field(default_factory=list)
+    registry_name: str = "mongodb_batch"
+
+    def __post_init__(self):
+        # Until python3.8 is deprecated, this is a limitation of dataclass inheritance
+        # to make it a required field
+        if len(self.list_of_ids) == 0:
+            raise ValueError("list_of_ids is required")
+
+    @property
+    def unique_id(self) -> str:
+        return ",".join(sorted(self.list_of_ids))
+
+    @requires_dependencies(["pymongo"], extras="mongodb")
+    def _get_docs(self):
+        """Fetches all documents in a collection."""
+        from bson.objectid import ObjectId
+
+        # I don't like this here. Would prefer to get client
+        # from MongoDBSourceConnector or SimpleMongoDBConfig
+        client = self.connector_config.generate_client()
+        collection = self.connector_config.get_collection(client)
+        # MondoDB expects a list of ObjectIds (yet won't take a list comprehension for some reason
+        list_of_object_ids = []
+        for x in self.list_of_ids:
+            list_of_object_ids.append(ObjectId(x))
+        return list(collection.find({"_id": {"$in": list_of_object_ids}}))
+
+    def get_files(self):
+        documents = self._get_docs()
+        for doc in documents:
+            ingest_doc = MongoDBIngestDoc(
+                processor_config=self.processor_config,
+                read_config=self.read_config,
+                connector_config=self.connector_config,
+                document_meta=MongoDBDocumentMeta(
+                    collection=self.connector_config.collection,
+                    document_id=str(doc.get("_id")),
+                    date_created=doc.get("_id").generation_time.isoformat(),
+                ),
+                document=doc,
+            )
+            ingest_doc.update_source_metadata()
+            del doc["_id"]
+            filename = ingest_doc.filename
+            flattened_dict = flatten_dict(dictionary=doc)
+            str_values = [str(value) for value in flattened_dict.values()]
+            concatenated_values = "\n".join(str_values)
+
+            filename.parent.mkdir(parents=True, exist_ok=True)
+            with open(filename, "w", encoding="utf8") as f:
+                f.write(concatenated_values)
+
+            self.ingest_docs.append(ingest_doc)
+
+
+@dataclass
+class MongoDBSourceConnector(SourceConnectorCleanupMixin, BaseSourceConnector):
+    connector_config: SimpleMongoDBConfig
     _client: t.Optional["MongoClient"] = field(init=False, default=None)
 
```

Comment:
please define a to_dict method and in it remove this _client since it won't be serializable. [Here](https://github.com/Unstructured-IO/unstructured/pull/2399)is a reference PR to where this has been done for another connector